src/plugins/discover/public/context_awareness/profile_providers/security/security_root_profile/profile.tsx

```diff
@@ -7,25 +7,80 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import React, { FunctionComponent, useMemo } from 'react';
+import { CustomCellRenderer, DataGridCellValueElementProps } from '@kbn/unified-data-table';
 import { RootProfileProvider, SolutionType } from '../../../profiles';
 import { ProfileProviderServices } from '../../profile_provider_services';
 import { SecurityProfileProviderFactory } from '../types';
+import { createCellRendererAccessor } from '../accessors/get_cell_renderer_accessor';
+import { createAppWrapperAccessor } from '../accessors/get_app_wrapper_accessor';
+
+interface SecurityRootProfileContext {
+  store?: unknown;
+  appWrapper?: (props: { children?: React.ReactNode }) => JSX.Element;
+  getCellRenderer?: (
+    fieldName: string
+  ) => FunctionComponent<DataGridCellValueElementProps> | undefined;
+}
 
 export const createSecurityRootProfileProvider: SecurityProfileProviderFactory<
-  RootProfileProvider
-> = (services: ProfileProviderServices) => ({
-  profileId: 'security-root-profile',
-  isExperimental: true,
-  profile: {
-    getCellRenderers: (prev) => (params) => ({
-      ...prev(params),
-    }),
-  },
-  resolve: (params) => {
-    if (params.solutionNavId === SolutionType.Security) {
-      return { isMatch: true, context: { solutionType: SolutionType.Security } };
-    }
+  Promise<RootProfileProvider<SecurityRootProfileContext>>
+> = async (services: ProfileProviderServices) => {
```

Comment:
> [!WARNING]
> As far as I see this function doesn't need to be async as it's not awaiting any async process in its scope, removing this would allow you to revert the changes on the root profiles creation you added [here](https://github.com/elastic/kibana/pull/199279/files#diff-9b1628b4730e65fdda3fa437b658119e0394fbed546a9e110836c91ca9e05c11).

Reply:
Fixed in 6a8a7d9bc96. It was an artifact for when it was actually `async`. Thanks for spotting it @tonyghiani .